x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java

```diff
@@ -146,12 +146,16 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         XContentSource that = (XContentSource) o;
-        return Objects.equals(bytes, that.bytes) &&
-            contentType == that.contentType;
+        return Objects.equals(data(), that.data());
```

Comment:
what is this part of the change doing? Was this an existing bug you found? Or did this help facilitate making the round trip test easier (the `package org.elasticsearch.protocol.xpack.watcher` test).

Reply:
This was for the round-trip tests; if the internal map got shuffled before it was serialized, then the BytesReferences would compare as different even though their semantic content was the same.